core/src/main/java/org/elasticsearch/index/query/PercolatorQuery.java

```diff
@@ -31,43 +30,47 @@
 import org.apache.lucene.search.Scorer;
 import org.apache.lucene.search.TwoPhaseIterator;
 import org.apache.lucene.search.Weight;
-import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.lucene.Lucene;
-import org.elasticsearch.index.fieldvisitor.SingleFieldsVisitor;
-import org.elasticsearch.index.mapper.internal.UidFieldMapper;
 import org.elasticsearch.index.percolator.ExtractQueryTermsService;
 
 import java.io.IOException;
-import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
 
 import static org.apache.lucene.search.BooleanClause.Occur.FILTER;
 import static org.apache.lucene.search.BooleanClause.Occur.MUST;
 
-final class PercolatorQuery extends Query {
+public final class PercolatorQuery extends Query {
```

Comment:
can you make it implement Accountable? The reasoning is that the query cache will take the memory of this query into account if it implements Accountable, which I think could be useful since the source field can potentially be large?